simpletransformers/t5/t5_model.py

```diff
@@ -42,7 +42,7 @@ def chunks(lst, n):
 
 class T5Model:
     def __init__(
-        self, model_name, args=None, use_cuda=True, cuda_device=-1, **kwargs,
+        self, model_name=None, args=None, config=None, tokenizer=None, use_cuda=True, cuda_device=-1, **kwargs,
```

Comment:
I'd be more comfortable with keeping `model_name` as a required parameter which the user can set to `None`. The idea is that you have to explicitly use a non-pretrained model if you need to.

`config` can already be passed as part of `args` so I don't think we need to add a separate parameter for it.

```suggestion
        self, model_name, args=None, tokenizer=None, use_cuda=True, cuda_device=-1, **kwargs,
```

Reply:
Yes, I totally agree with you, maybe it can be specified in the docs that a "non-pretrained" version can be set with `None` tag.